apis/python/examples/soma-slice-query.md

```diff
@@ -0,0 +1,200 @@
+Here we show an example of doing a _slice query_ across a `SOMACollection`.
+
+## Populate the collection
+
+Here we use a few small sample files included in this repository.
+
+```
+import tiledbsc
+import tiledbsc.io
+import os
+import shutil
+
+soco_path = './soco-attribute-filter'
+if os.path.exists(soco_path):
+    shutil.rmtree(soco_path)
+
+soco = tiledbsc.SOMACollection(soco_path)
+if not soco.exists():
+    soco._create()
+
+for name, h5ad in [
+    ('subset-soma-01', './anndata/subset-soma-01.h5ad'),
+    ('subset-soma-02', './anndata/subset-soma-02.h5ad'),
+    ('subset-soma-03', './anndata/subset-soma-03.h5ad'),
+    ('subset-soma-04', './anndata/subset-soma-04.h5ad'),
+]:
+    soma_path = os.path.join(soco_path, name)
+    soma = tiledbsc.SOMA(soma_path)
+    tiledbsc.io.from_h5ad(soma, h5ad)
```

Comment:
I would argue that this is a non-pythonic way to call a function with such a name. Generally, I would expect that a `from_*` function returns the created object, instead of mutating an argument. I understand that the object is disk-based so it makes sense to pre-create it and pass it as an argument, but then I'd probably rename the function to something that makes the side effect more explicit (like `write_from_h5ad`). @bkmartinjr your opinion?

Reply:
@ebezzi originally it was `soma.from_h5ad(h5ad)` but there was a desire to have a separate `tiledbsc.io` namespace (#83) in order to more clearly separate I/O (peripheral) from SOMA-API processing (central).

Reply:
I'm not sure if it is non-pythonic (I lack expertise to make that determination).  But I do find it awkward and likely error-prone as the two step process (create the soma, write the soma) allows for cases that will fail (eg, the soma is modified before `from_h5ad` is called).

I personally prefer a single function that both creates the soma and ingests the H5AD, eg,   `soma.io.from_h5ad(soma_path: str, h5ad_path: str, ...) -> soma`.   Ie, this function would call the SOMA constructor.

Reply:
good to know @bkmartinjr -- thanks for the feedback! I'm fine with this; only wish I'd known sooner :)

Reply:
@bkmartinjr @ebezzi I'll go ahead & make this change but on a separate PR -- this `soma.io` business is old & predates this PR, and this change can be neatly factored out

Reply:
Maybe it's worth posting this as a separate issue for discussion/feedback? The R API currently works this way too (ie, ingestion and SOMA instantiation are separate steps) and altering that behavior would require a lot of small updates to docs/tests/etc.